### DISCUSSION TITLE
Platform-specific GDExtension support and IfDef in GDScript so that we can exclude platform-specific GDExtension lines

### DISCUSSION THREAD

**Discussion by Anutrix** [godotengine/godot-proposals#9640] [Scripting]
This is 2 parter request. I am facing this issue where I want to run some C++ code only when a Godot Project is exported on Windows.

1. Currently, there doesn't seem to be way to disable a GDExtension plugin for specific platforms. For example, if don't compile and add Android platform we get:
```
No suitable library found for GDExtension: res://bin/win_temp_avatar_trigger.gdextension. Possible feature flags for your platform: mobile, android, etc2, astc, arm64, template, debug, template_debug, single
```
I feel like this is something that can be fixed in export settings exclude section.

2. Complementing above, we need `ifdef`/`ifdefined` for GDScript so that lines that are only present for platform-specific GDExtension plugin provided Nodes don't cause 'Undefined/Unknown Class' error. Exact error currently seen:
```
Parser Error: Identifier "WinTempAvatarTrigger" not declared in the current scope.
```

**Comment by Calinou** [comment]
`#if` for GDScript is already being tracked in https://github.com/godotengine/godot-proposals/issues/7375.
